gemfirexd/client/src/main/java/com/pivotal/gemfirexd/internal/client/am/Statement.java

```diff
@@ -2715,6 +2715,11 @@ void parseSqlAndSetSqlModes(String sql) throws SqlException {
             firstToken = getStatementToken(sql, idx + 1);
           }
         }
+        if (firstToken.equalsIgnoreCase("set")) {
+          if (sql.toLowerCase().contains(" schema ")) {
+              this.connection_.setSchemaSql_ = sql;
+          }
+        }
 // GemStone changes END
```

Comment:
In my first comment the local/global got lost as tags:

If we could add int[] argument which is set to index if non-null, then can also use it for local/global handling that is being done above this which is also inaccurate since it just searches for '>' that can be trouble if such were to be inside comments for example.